feat(smolagents): updates to latest and makes examples use opentelemetry-instrument

codefromthecrypt opened the pull request:
This update smolagents code where recent versions drifted:
* [remove of `ManagedAgent`](https://github.com/huggingface/smolagents/pull/484)
* [removal of `max_tokens` from all models](https://github.com/huggingface/smolagents/pull/475)

This also polishes the examples directory and adds a README there, some of the notable changes below:

* use `opentelemetry-instrument`, so that you can easily copy/paste examples without needing to edit in instrumentation
* remove explicit code that is the same as openai default behavior
* add env.example to document important settings that apply to main or otel code
* added missing tests for `opentelemetry-instrument` entrypoint and `._tracer` is a `OITracer`

I can do make the main README, use opentelemetry-instrument, too if this seems alright.

codefromthecrypt commented:
fyi @mikeldking I tested all of these with phoenix. works fine "on my laptop" ™️

codefromthecrypt commented:
buffed out a couple other drifts and documented them in the PR desc

nate-mar commented:
Nice! you're on a roll @codefromthecrypt ! :smiling 

Can we leave at least one example, or perhaps add to the readme that you could instrument without opentelemtry-instrument, in case you need some specific customization. Correct me if I'm wrong, but I think there are certain things that you can't do with the no-code approach (batch vs simple span processor?).

codefromthecrypt commented:
hey @nate-mar, and thanks for the feedback!

I added a ENV var that exports spans every second. This should help for the higher latency examples, which before this would only flush when the program completes or 5s whichever comes first.

I also added a note to the README about what to copy/paste alternatively.

Next q is should I make corresponding updates to the main smolagents instrumentation README? It currently has manual tracing setup there. Besides fuzz I've left that one alone so far.

nate-mar commented:
Awesome, thanks! This is great!

Sure, go for it re: updating the main smolagents readme ! I would only just mention the same thing about including a blurb/section about the manual config for additional flexibility if needed. and noted about the metrics!

codefromthecrypt commented:
Ok I pared down the instructions to least necessary. The top-level example needs no special env variables, due to not using openai etc, so that's a bit simpler. PTAL!

codefromthecrypt commented on python/instrumentation/openinference-instrumentation-smolagents/README.md:
```diff
@@ -14,42 +14,57 @@ pip install openinference-instrumentation-smolagents
 ## Quickstart
 
-This quickstart shows you how to instrument your guardrailed LLM application 
+This quickstart shows you how to instrument your LLM agent application.
```
I noticed this copy/paste elsewhere, so will try to scrub for that if I end up nudging another package.

codefromthecrypt commented on python/instrumentation/openinference-instrumentation-smolagents/tests/openinference/instrumentation/smolagents/test_instrumentor.py:
```diff
@@ -95,6 +95,15 @@ def openai_api_key(monkeypatch: pytest.MonkeyPatch) -> str:
     return api_key
 
 
+class TestInstrumentor:
```
added this to make sure things don't drift and we lose registration by accident

codefromthecrypt commented on python/instrumentation/openinference-instrumentation-smolagents/tests/openinference/instrumentation/smolagents/test_instrumentor.py:
```diff
@@ -95,6 +96,19 @@ def openai_api_key(monkeypatch: pytest.MonkeyPatch) -> str:
+        assert isinstance(instrumentor, SmolagentsInstrumentor)
+
+    # Ensure we're using the common OITracer from common openinference-instrumentation pkg
+    def test_oitracer(self, instrument: Any) -> None:
```
shouldn't need this with autouse

```suggestion
    def test_oitracer(self) -> None:
```

nate-mar reviewed (approved):
Amazing! thanks again!

codefromthecrypt commented:
thanks! great timing as I'm online temporarily today and tomorrow
